sshspawner/sshspawner.py

```diff
@@ -99,16 +107,41 @@ def clear_state(self):
     async def start(self):
         """Start single-user server on remote host."""
 
+        username = self.user.name
+        kf = self.ssh_keyfile.format(username=username)
+        cf = kf + "-cert.pub"
+        k = asyncssh.read_private_key(kf)
+        c = asyncssh.read_certificate(cf)
+
         self.remote_host = self.choose_remote_host()
         
         self.remote_ip, port = await self.remote_random_port()
         if self.remote_ip is None or port is None or port == 0:
             return False
+        self.remote_port = str(port)
         cmd = []
 
         cmd.extend(self.cmd)
         cmd.extend(self.get_args())
 
+        with TemporaryDirectory() as td:
```

Comment:
One other thing; this should be blocked off if internal ssl is not set.  I think that's a temporary fix here.  Once this is merged we can move it around and get rid of the multi-host stuff to make it work with move certs hook.  Should have caught this yesterday.

Reply:
added conditional and tested locally, should be good to go. thanks for the thorough review!